src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -84,7 +84,15 @@ namespace
         }
         void operator() (const UUID & x) const
         {
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+            auto tmp_x = x.toUnderType();
+            char * start = reinterpret_cast<char *>(&tmp_x);
```

Comment:
I tried to use std::byteswap(), however it cannot compile since it only supports native integer, and use it on DB:UInt128 will emit the following error:
```
'wide::integer<128, unsigned int>' does not satisfy 'integral'
template <integral _Tp>
```

Reply:
I see. Okay than and thanks for trying.